### PR TITLE
Add option to restart machine deployments

### PR DESCRIPTION
**Pull request by maciaszczykm**
### Which issue(s) this PR fixes
Blocked by https://github.com/kubermatic/kubermatic/pull/7340.
Fixes https://github.com/kubermatic/dashboard/issues/3440.

### Release note
```release-note
Added option to restart machine deployments.
```

### CODE REVIEW

**Comment by maciaszczykm** [comment]
/hold 

To wait for API.

/assign @kgroschoff @floreks

**Comment by maciaszczykm** [comment]
/hold cancel

API is ready.

File: src/app/cluster/cluster-details/machine-deployment-details/template.html [hunk 1]
```diff
@@ -29,6 +29,12 @@
       <div fxFlex></div>
       <div fxLayout="row"
            fxLayoutGap="10px">
+        <button mat-icon-button
+                color="tertiary"
+                (click)="showRestartDialog()"
+                [disabled]="!isEditEnabled()">
+          <i class="km-icon-reset"></i>
```
Comment (kgroschoff) [inline src/app/cluster/cluster-details/machine-deployment-details/template.html:36]: Not sure about this one, and we can also wait for @cschieder to be back, but maybe `km-icon-regenerate` would be a better fit? 
We're already using it in the serviceaccounts > tokens to regenerate the token. Regenerate for me feels more like a restart, than reset 🤔 But maybe I'm wrong 😅

**Review by kgroschoff — COMMENTED** [review]
Just a small comment

File: src/app/cluster/cluster-details/machine-deployment-details/template.html (hunk 1 above)
Comment (floreks) [reply src/app/cluster/cluster-details/machine-deployment-details/template.html:36]: This is a valid point. Actually, to me the `reset` icon looks like a `Go back` button, not a real `reset` :slightly_smiling_face:.
I'd go with the `regenerate` icon also.

Comment (maciaszczykm) [reply src/app/cluster/cluster-details/machine-deployment-details/template.html:36]: True. I didn't remember about this icon. Fixed.

**Review by kgroschoff — COMMENTED** [review]
/lgtm

**Pull request merged**